csharp/src/Apache.Arrow/Types/StructType.cs

```diff
@@ -28,7 +28,7 @@ public sealed class StructType : ArrowType
 
         public IEnumerable<Field> Fields => _fields;
```

Comment:
Since we are deriving from `NestedType` now, why do we need the `Fields` property? `NestedType` already has a `Children` property, which will be the same thing.

Reply:
Note that C++ standardized on the "Field" name: https://github.com/apache/arrow/pull/7132

Reply:
Removed `_fields`. I'll open another PR to rename all instances of Children with Field (there are 80+ hits, so I'd rather not pollute this PR).

Reply:
I resolved this by moving `Fields` to the base `NestedType` and obsoleting `Children`.